client/netdb.c

```diff
@@ -133,13 +134,14 @@ static struct host *random_host(int mask)
 
 static struct host *ipport2host(struct host *h, const char *ipport, int flags)
 {
-	unsigned ip[5];
+	unsigned ip[4];
+	unsigned short port;
 
-	if (sscanf(ipport, "%u.%u.%u.%u:%u", ip, ip + 1, ip + 2, ip + 3, ip + 4) != 5
-		|| (ip[0] | ip[1] | ip[2] | ip[3]) & ~0xff || ip[4] & ~0xffff) return 0;
+	if (sscanf(ipport, "%u.%u.%u.%u:%u", &ip[0], &ip[1], &ip[2], &ip[3], &port) != 5
```

Comment:
Why do you use unsigned int for ip that is smaller and unsigned short for port that is bigger?
Why don't you use uint8_t for ip and uint16_t for port?